juniper_official/Node-Exporter/check-cpu-usage.rule

```diff
@@ -0,0 +1,164 @@
+            sensor node-cpu {
+                /*
+                 * Sensor configuration to get data from network devices.
+                 */			
+                description "node cpu in seconds";
+                server-monitoring {
+                    sensor-name /node/cpu;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined by specifying the sensor paths
+             */			
+            field cpu {
+                sensor node-cpu {
+                    path cpu;
+                }
+                type string;
+                description "cpu number";
+            }
+            field cpu-total-seconds {
+                sensor node-cpu {
+                    path /node/cpu/seconds/total;
+                    zero-suppression;
+                }
+                type integer;
+                description "total cpu seconds";
+            }
+            field cpu-used-percentage {
+                formula {
+                    user-defined-function {
+                        function-name used-percentage;
+                        argument total "$diff-prev-curr";
+                        argument used "$frequency";
+                    }
+                }
+                type float;
+                description "CPU used percentage for a specific CPU number and mode";
+            }
+            field diff-prev-curr {
+                formula {
+                    user-defined-function {
+                        function-name difference;
+                        argument key_name "$cpu";
+                        argument sub_key_name "$mode";
+                        argument value "$cpu-total-seconds";
+                    }
+                }
+                type integer;
+                description "Difference in current and previous cpu total seconds value";
+            }
+            field frequency {
+                constant {
+                    value "{{freq}}";
+                }
+                type integer;
+                description "time interval to calculate cpu used percentage";
+            }
+            field mode {
+                sensor node-cpu {
+                    path mode;
+                }
+                type string;
+                description "Different modes like iowait,idle,nice,system etc";
+            }
+            field threshold {
+                constant {
+                    value "{{threshold}}";
+                }
+                type integer;
+                description "threshold value to compare cpu used percentage";
+            }
+            /*
+             * Anomaly detection logic.
+             */			
+            trigger cpu-usage {
+                frequency 1offset;
+                term above-threshold {
+                    when {
+                        greater-than-or-equal-to "$cpu-used-percentage" "$threshold" {
+                            time-range 3offset;
+                            any;
```

Comment:
Don't we want to flag the cpu-usage KPI as red only if the CPU is pegged over a period of time - in which case shouldn't it be all instead of any?

Another option could be to mark this as a yellow trigger (for any) and have a red trigger for all. So long as we don't inundate the operator with too many yellow triggers, even this might be acceptable.

Reply:
Changed to yellow.